src/wcl/tracing.h

```diff
@@ -59,6 +58,20 @@ struct Event {
 
     return &it->second;
   }
+
+  Event message(const char* fmt, ...) && __attribute__((format(printf, 2, 3)));
+  Event message(const char* fmt, va_list args) &&;
+
+  Event urgent() && __attribute__((warn_unused_result));
+  Event time() && __attribute__((warn_unused_result));
+  Event pid() && __attribute__((warn_unused_result));
+
+  Event info() && __attribute__((warn_unused_result));
+  Event warning() && __attribute__((warn_unused_result));
+  Event error() && __attribute__((warn_unused_result));
```

Comment:
Instead of these, we should instead have `level` that takes a string as input. These won't generally be used so I think a more orthogonal, generic method is better.